### PR TITLE
En Passan and Castling added

### PR DESCRIPTION
**Pull request by The-Chel**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -25,15 +25,21 @@
...
     </div>
     <div>
       <canvas id="myCanvas" height="850px" width="850px" style=" display: inline-block; margin: 10px 0;"></canvas>
       <div style="display: inline-block; vertical-align: top;">
         <span id="turnCount" style="font-size: 30px;">White turn!</span> <br>
 
         <span style="font-size: 45px;">History</span> <br>
-        <div id="historyHolder"></div>
+        <div>
+          <span style="margin-right: 25px;">White</span>
```
Comment (The-Chel) [inline index.html:37]: Will add flex box and columns later

File: src/index.js [hunk 2]
```diff
@@ -63,34 +59,119 @@ function figureAdd (id, color, type, specialFeature) {
...
+            if (getFigureById(removeId).color !== 'black') {
+              figureRemove(removeId);
+            }
+          }
+          if (check) {
+            addHistory('capture', fromId, toId);
+            check = false;
+          }
+        }
+      } catch (error) {}
```
Comment (drunkevich) [inline src/index.js:138]: why do you need this?:)

File: src/index.js [hunk 3]
```diff
@@ -365,13 +494,11 @@ function figureMove (idIn) {
 
         if (firstColor !== secondColor && boardSquare[squareId].canMove) {
           console.log('by enemy');
-          figureRemove(squareId);
+          // figureRemove(squareId);
```
Comment (drunkevich) [inline src/index.js:497]:
```suggestion
```

File: src/index.js [hunk 4]
```diff
@@ -365,13 +494,11 @@ function figureMove (idIn) {
...
           figurePositionChange(squareId, idIn);
         } else { console.log('by ally'); }
       }
-    } catch (error) {
-
-    }
+    } catch (error) { console.log('ERROR in figureMove()'); }
```
Comment (drunkevich) [inline src/index.js:501]: kinda useless catch

File: src/index.js [hunk 5]
```diff
@@ -517,6 +656,20 @@ function highlightMove (id) {
         }
       } catch (e) {}
```
Comment (drunkevich) [inline src/index.js:657]: ?

File: src/index.js [hunk 6]
```diff
@@ -517,6 +656,20 @@ function highlightMove (id) {
...
       // En Pasant
+      try {
+        if (boardSquare[rightId].isEmpty === false && getFigureById(rightId).enPassant === true) {
+          if (element.color === 'white') {
+            movesDraw(topRightId, 'top-right', element, 1);
+          } else movesDraw(bottomRightId, 'bottom-right', element, 1);
+        }
+      } catch (e) {}
```
Comment (drunkevich) [inline src/index.js:665]: but why???

Comment (The-Chel) [reply src/index.js:665]: Sometimes 'rightId' may not exist and boardSquare[rightId].isEmpty will lead to an error: Trying to find ''isEmpty' in undefined

File: src/index.js (hunk 4 above)
Comment (The-Chel) [reply src/index.js:501]: Same here, sometimes boardSquare[SOMETHING] will not have and element in it, and trying accessing any property will lead to and error

File: src/index.js (hunk 6 above)
Comment (drunkevich) [reply src/index.js:664]: so maybe
`if (rightId && boardSquare[rightId].isEmpty === false && getFigureById(rightId).enPassant === true) {`

File: src/index.js [hunk 7]
```diff
@@ -283,38 +403,89 @@ function figDef () {
...
+      pushValue = fromId[0] + ':' + toId + 'Q';
+      break;
+    case 'promotion':
+      pushValue = toId + 'Q';
+      break;
+    case 'castling-queen':
+      if (color === 'white') pushValue = '0-0-0';
+      break;
+    case 'castling-king':
+      if (color === 'white') pushValue = '0-0';
```
Comment (The-Chel) [inline src/index.js:453]:
```suggestion
      pushValue = '0-0';
```

File: src/index.js [hunk 8]
```diff
@@ -283,38 +403,89 @@ function figDef () {
...
+      }
+      break;
+    case 'capture-promotion':
+      pushValue = fromId[0] + ':' + toId + 'Q';
+      break;
+    case 'promotion':
+      pushValue = toId + 'Q';
+      break;
+    case 'castling-queen':
+      if (color === 'white') pushValue = '0-0-0';
```
Comment (The-Chel) [inline src/index.js:450]:
```suggestion
      pushValue = '0-0-0';
```

**Pull request merged**